### PR TITLE
Fixes a few simple animal huds

### LINKED ISSUE #17434: Medical HUD healthbar is displayed on invisible revenants
This clearly reveals revenants who are supposed to not be seen at all

### PR DESCRIPTION
**Pull request by ChangelingRain**
:cl: Joan
rscadd: Medical huds will now show parasites in dead creatures.
/:cl:

Revenants, statues, and ascended shadowlings have no medical hud now.
Morphs have no medical hud while morphed.
Fixes medical hud not displaying crit states properly on critted aliens.
Fixes #17434

### CODE REVIEW

File: code/game/data_huds.dm [hunk 1]
```diff
@@ -70,7 +70,10 @@
 /proc/RoundHealth(mob/living/M)
 	if(M.stat == DEAD)
 		return "health-100" //what's our health? it doesn't matter, we're dead
-	var/resulthealth = (M.health / M.maxHealth) * 100
+	var/maxhealth = M.maxHealth
+	if(iscarbon(M) && M.health < 0)
+		maxhealth = 100 //so crit shows up right for aliens and other high-health carbon mobs; noncarbons don't have crit.
```
Comment (ChangelingRain) [inline code/game/data_huds.dm]: I hate this but ~oh well~

**Comment by LatD** [comment]
I'm not sure if this is worth a bug report or if this is getting fixed already but earlier today I saw humans polymorphed to chicks (the chick simple mob) and the health hud showed them nearly in crit even though they had max hp. Probably due to their very small max hp.

**Comment by ChangelingRain** [comment]
The HUD display is based off of health divided by maxhealth, and that specific bug is unrelated; the chick simple animal has 1 health, but doesn't have a maxhealth set so it defaults to 20. This only fucks up the medical hud, not damage.

**Comment by LatD** [comment]
Oh it doesn't start with max hp. Then it's not related to medhuds or this PR in any way but that seems really weird.

..now that I look the simple animals they all seem to have defined healths like that 1 and up to 50 with cows but they all still have maxhealth defaulted at 20. I'll make a separate issue for this.

**Comment by GunHog** [comment]
While you are at it, know that the large, royal xenos seem to have an odd location for their bars.

**Comment by ChangelingRain** [comment]
Yeah I know, not totally sure on the best way to fix it.

File: code/game/data_huds.dm (hunk 1 above)
Comment (KorPhaeron) [inline code/game/data_huds.dm]: This is pretty gross

**Pull request merged**